app/controllers/restaurants_controller.rb

```diff
@@ -13,6 +14,36 @@ def new
     @restaurant = Restaurant.new
   end
 
+  def edit
+    @restaurant = Restaurant.find(params[:id])
+    if @restaurant.owner?(current_user)
+      @restaurant
+    else
+      flash[:notice] = "Your account is not authorized to view this page"
+    end
+  end
+
+  def update
+    @restaurant = Restaurant.update(params[:id], restaurant_params)
+    if @restaurant.save
+      flash[:notice] = "Restaurant saved successfully!"
+      redirect_to @restaurant
+    else
+      flash[:notice] = "Update failed"
+      render :new
+    end
+  end
+
+  def destroy
+    @restaurant = Restaurant.find(params[:id])
+    if @restaurant.owner?(current_user)
+      @restaurant.destroy
+      redirect_to restaurants_path, notice: 'Restaurant deleted successfully!'
+    else
+      flash[:notice] = "Your account is not authorized to view this page"
+    end
+  end
```

Comment:
Same comment as above about either showing user abilities or not based on their privileges in the views rather than the controller telling them so.